### PR TITLE
Cleanup search sub transport actions and collapse o.e.action.search.type package into o.e.action.search

### LINKED ISSUE #11710: Shield duplicates _search api audits
A standard GET by ID results in a single log entry in the audit log. Search (_search) results in two identical log entries. Every time. 

* Default ES 1.6.0 installation. 
* Added latest Shield. 
* Added "search_admin" user. 

Everything works great out of the box except I get _search audits duplicated in the log. ONLY "SearchRequest" audits. I haven't found any other api yet which results in this strange behavior.

```
[2015-06-16 18:37:24,705] [esdev-shieldpoc01] [transport] [access_granted]      origin_type=[rest], origin_address=[/10.30.24.36:55308], principal=[search_admin], action=[indices:data/read/search], indices=[test], request=[SearchRequest]
[2015-06-16 18:37:24,705] [esdev-shieldpoc01] [transport] [access_granted]      origin_type=[rest], origin_address=[/10.30.24.36:55308], principal=[search_admin], action=[indices:data/read/search], indices=[test], request=[SearchRequest]
```

### PR DESCRIPTION
**Pull request by javanna**
TransportSearchTypeAction and subclasses are not actually transport actions, but just support classes useful for their inner async actions that can easily be extracted out so that we get rid of one too many level of abstraction.

Same pattern can be applied to TransportSearchScrollQueryAndFetchAction & TransportSearchScrollQueryThenFetchAction which we could remove in favour of keeping only their inner classes named SearchScrollQueryAndFetchAsyncAction and SearchScrollQueryThenFetchAsyncAction.

Remove org.elasticsearch.action.search.type package, collapsed remaining classes into existing org.elasticsearch.action.search package

Make also ParsedScrollId ScrollIdForNode and TransportSearchHelper classes and their methods package private.

Closes #11710

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/action/search/SearchDfsQueryThenFetchAsyncAction.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            DfsSearchResult dfsResult = entry.value;
+            DiscoveryNode node = nodes.get(dfsResult.shardTarget().nodeId());
+            QuerySearchRequest querySearchRequest = new QuerySearchRequest(request, dfsResult.id(), dfs);
+            executeQuery(entry.index, dfsResult, counter, querySearchRequest, node);
+        }
+    }
+
+    void executeQuery(final int shardIndex, final DfsSearchResult dfsResult, final AtomicInteger counter,
+                      final QuerySearchRequest querySearchRequest, final DiscoveryNode node) {
+                            searchService.sendExecuteQuery(node, querySearchRequest, new ActionListener<QuerySearchResult>() {
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/action/search/SearchDfsQueryThenFetchAsyncAction.java]: indentation looks off here

**Comment by jaymode** [comment]
left a minor comment about formatting, but other than that LGTM

**Comment by jpountz** [comment]
+1 I like it

**Comment by javanna** [comment]
I pushed this to master and backported to 2.x as it fixes #11710. The backport required some work given that scan and count search type were removed in master, plus some other conflict. I haven't backported to 2.2, @clintongormley should I look into it or is 2.3 good enough?

**Comment by clintongormley** [comment]
I'm good with 2.3 - it's too big a change for a bugfix release I think

**Comment by pciccarese** [comment]
Hi, I see #11710 is not going to be fixed in 2.2.0 but is it fixed in 2.3.0? 
It looks like I am suffering the ActionFilter.Simple double invocation on 2.2.0.
Thank you!

**Comment by javanna** [comment]
Hi @pciccarese you are correct. This PR was the fix for #11710 and will go out with 2.3. We decided not to include the fix in a bugfix release as it involved a refactoring that touched quite some classes, too big to be considered just a bugfix.

**Comment by pciccarese** [comment]
Thank you @javanna , do you know if, besides 2.2.0, other previous versions were effected by that bug?

**Comment by javanna** [comment]
> do you know if, besides 2.2.0, other previous versions were effected by that bug?

Every single version before 2.3.0 is affected by it, from the very first one, sorry! :)

**Pull request merged**